pkg/resource/ignored.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 Upbound Inc.
+*/
+
+package resource
+
+import (
+	"fmt"
+	"sort"
+)
+
+// GetIgnorableFields returns a sorted list of ignorable fields by looking for
+// differences between the `initProvider` and `forProvider` maps. The ignored
+// fields are the ones that are present in initProvider, but not in forProvider.
```

Comment:
Should we error out when the same field is defined both in `initProvider` and `forProvider` or silently ignore the `initProvider` one by overriding it with the one from `forProvider` ? It looks like you implemented the latter, but wondering if we should rather error out considering there is no reason to have such a configuration.

Reply:
ideally if the same field is defined in both, we would just disregard the 'initProvider' field. But as we use mergo.Merge, to achieve the goal of merging maps and arrays, we use the option mergo.WithSliceDeepCopy which includes overrides. I wanted to play a bit more with it to see if we can get the desired effect, but not sure if possible.

If we error out, it would fix that problem at least :D. Not sure which is the better UX though

Reply:
I cannot think of a meaningful reason behind defining a parameter in both places. But, if that happens, one could expect the one under `forProvider` should override the one under `initProvider`. I am fine as long as we could implement one of the two options.